harness/determined/exec/tensorboard.py

```diff
@@ -161,17 +181,112 @@ def start_tensorboard(
+                t = threading.Timer(READY_SIGNAL_DELAY, self._emit_ready_signal)
+                t.start()
+            self._num_fetched_files += 1
+
+    def _emit_ready_signal(self) -> None:
+        print(TENSORBOARD_TRIGGER_READY_MSG, flush=True)
+
+    def get_num_fetched_files(self) -> int:
+        with self._lock:
+            return self._num_fetched_files
+
+
+class TBFetchIterationThread(threading.Thread):
+    """Thread to continuously iterate over the fetchers files and add them to a threading.Queue
+
+    Note: We are making the assumption that there will only be one of these running per process.
+    If we add more, then the base fetcher will need to support locking around the _file_records
+    dictionary. Defined in <ROOT>/harness/determined/tensorboard/fetchers/base.py
+    """
+
+    def __init__(
+        self,
+        fetcher: fetchers.Fetcher,
+        work_queue: queue.Queue,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        self._fetcher = fetcher
+        self._work_queue = work_queue
+        super().__init__(*args, **kwargs)
+
+    def run(self) -> None:
+        while True:
+            try:
+                for filepath in self._fetcher.list_all_generator():
+                    self._work_queue.put(filepath, block=True)
+            except requests.Timeout as e:
+                logging.warning(
+                    f"Timeout listing TensorBoard files from {self._fetcher}. Error: {e}"
+                    " (retrying in {FULL_ITERATION_SLEEP_TIME}s)...",
+                    exc_info=True,
+                )
+            except Exception as e:
+                logging.warning(
+                    f"Failure listing TensorBoard files from {self._fetcher}. Error: {e}"
+                    f" (retrying in {FULL_ITERATION_SLEEP_TIME}s)...",
+                    exc_info=True,
+                )
+            finally:
+                time.sleep(FULL_ITERATION_SLEEP_TIME)
+
+
+class TBFetchThread(threading.Thread):
+    """Thread to continuously read from the queue and fetch the files"""
+
+    def __init__(
+        self,
+        fetcher: fetchers.Fetcher,
+        work_queue: queue.Queue,
+        new_file_callback: Callable,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        self._fetcher = fetcher
+        self._work_queue = work_queue
+        self._new_file_callback = new_file_callback
+        super().__init__(*args, **kwargs)
+
+    def run(self) -> None:
+        while True:
+            try:
+                filepath = self._work_queue.get(block=True)
+                self._fetcher._fetch(filepath, self._new_file_callback)
+            except requests.Timeout as e:
+                logging.warning(
+                    f"Timeout fetching TensorBoard files from {self._fetcher}. Error: {e}"
+                    f" (retrying)...",
+                    exc_info=True,
+                )
+            except Exception as e:
+                logging.warning(
```

Comment:
nit: this is the same log as the above checked exception. maybe we should just catch the generic exception and log it? seems weird to catch just the Timeout exception and not other checked exceptions.

Reply:
That's fine. I edited to do that instead